doc/api/vm.md

```diff
@@ -1616,6 +1614,156 @@ inside a `vm.Context`, functions passed to them will be added to global queues,
+
+If this option is not specified, or if it's undefined, code containing
+`import()` can still be compiled by the vm APIs, but when the compiled code is
+executed and it actually calls `import()`, the result will reject with
+[`ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING`][].
+
+### When `importModuleDynamically` is `vm.constants.USE_MAIN_CONTEXT_DEFAULT_LOADER`
+
+This option is currently not supported for `vm.SourceTextModule`.
+
+With this option, when an `import()` is initiated in the compiled code, Node.js
+would use the default ESM loader from the main context to load the requested
+module and return it to the code being executed.
+
+```mjs
+import { Script, constants } from 'node:vm';
+import { resolve } from 'node:path';
+import { writeFileSync } from 'node:fs';
+
+const data = JSON.stringify({ hello: 'world' });
+writeFileSync(resolve(import.meta.dirname, 'foo.json'), data);
+
+const script = new Script('import("./foo.json", { with: { type: "json" } })', {
+  filename: resolve(import.meta.dirname, 'test-with-default.js'),
+  importModuleDynamically: constants.USE_MAIN_CONTEXT_DEFAULT_LOADER,
+});
+const result = await script.runInThisContext();
+//  { default: { hello: 'world' } }
+console.log(result);
+```
+
+```cjs
+const { Script, constants } = require('node:vm');
+const { resolve } = require('node:path');
+const { writeFileSync } = require('node:fs');
+
+const data = JSON.stringify({ hello: 'world' });
+writeFileSync(resolve(__dirname, 'foo.json'), data);
+
+const script = new Script('import("./foo.json", { with: { type: "json" } })', {
+  filename: resolve(__dirname, 'test-with-default.js'),
+  importModuleDynamically: constants.USE_MAIN_CONTEXT_DEFAULT_LOADER,
+});
+//  { default: { hello: 'world' } }
+script.runInThisContext().then(console.log);
+```
+
+There are a few caveats with loading modules using the default loader in the
+main context:
+
+1. The module being resolved would be relative to the `filename` option passed
+   to `vm.Script` or `vm.compileFunction()`. The resolution can work with a
+   `filename` that's either an absolute path or a URL string.  If `filename` is
+   a string that's neither an absolute path or a URL string, or if it's
+   undefined, the resolution will be relative to the current working directory
+   of the process. In the case of `vm.createContext()`, the resolution is always
+   relative to the current working directory since this option is only used when
+   there isn't a referrer script or module.
+2. The module is loaded using the default loader from the main context, so using
+   the result in code executed in a different `vm.Context` can lead to
+   cross-context peculiarities.
+3. For any given `filename` that resolves to a specific path, once the process
+   manages to load a particular module from that path, the result may be cached,
+   and subsequent load of the same module from the same path would return the
+   same thing. There is currently no way to bypass the caching behavior.
```

Comment:
But if an URL can be passed, whether as a string or an instance, couldn’t the cache be bypassed by adding a query parameter? `?ts=${Date.now()}` say.

Reply:
Added a note about URL strings.